### ISSUE TITLE
Single step debugging support

### ISSUE THREAD

**Issue by SuperWangKai** [marcobambini/gravity#286]
DEBUG_STACK() only support stack print. It's much better if we can stop as any line of code.

**Comment by SuperWangKai** [comment]
Additionally, I wonder if auto-completion and code structure display can be supported?

**Comment by marcobambini** [comment]
Yes it can be supported, I am working on Visual Studio Code support.
